File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int answer = 0;
        
        /* 알파벳 개수 체크 */
        int[] alph = new int[128];
        for(int i=0;i<s.length();i++) {
            alph[s.charAt(i)]++;
        }
        
        for(int i=0;i<alph.length;i++) {
            if(alph[i] != 0) {
                answer += alph[i]/2*2; // 2개 이상이면 가능한 짝수 개수 만큼 추가
                
                /* answer 글자 수가 짝수이면 나머지 알파벳 추가*/
                if(answer%2==0 && alph[i]%2!=0) answer++;
            }
        }
        return answer;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.py
from collections import Counter

class Solution:
    def longestPalindrome(self, s: str) -> int:
        alph_cnt = Counter(s)
        res = 0
        odd_flag = False

        for n in alph_cnt.values():
            res += n // 2 * 2
            if n % 2 != 0:
                odd_flag = True

        return res + (1 if odd_flag else 0)